src/mat.rs

```diff
@@ -1094,6 +1114,16 @@ macro_rules! mat4s {
 
 mat4s!(Mat4 => Rotor3, Bivec3, Vec4, Vec3, f32, Wat4 => WRotor3, WBivec3, Wec4, Wec3, f32x4);
 
+impl PartialEq for Mat4 {
+    fn eq(&self, other: &Self) -> bool {
+        self.cols[0] == other.cols[0] && self.cols[1] == other.cols[1] && self.cols[2] == other.cols[2] && self.cols[3] == other.cols[3]
+    }
+
+    fn ne(&self, other: &Self) -> bool {
+        self.cols[0] != other.cols[0] || self.cols[1] != other.cols[1] || self.cols[2] != other.cols[2] || self.cols[3] != other.cols[3]
```

Comment:
could you format these (and the rest) like

```rust
expression
|| nextexpr
|| nextexpr
```
etc

Reply:
now it's all formatted by rustfmt, oh and sorry for not doing it sooner, I am a bit sick